src/pages/projectSettings/tabs/ViewsAndFiltersTab/transformers.ts

```diff
@@ -0,0 +1,23 @@
+import { ProjectSettingsTabRoutes } from "constants/routes";
+import { ProjectSettingsQuery } from "gql/generated/types";
+import { FormToGqlFunction, GqlToFormFunction } from "../types";
+import { FormState } from "./types";
+
+type Tab = ProjectSettingsTabRoutes.ViewsAndFilters;
+
+export const gqlToForm: GqlToFormFunction<Tab> = (
+  data: ProjectSettingsQuery["projectSettings"]
+) => {
+  if (!data) return null;
+
+  return {};
+};
+
+export const formToGql: FormToGqlFunction<Tab> = (
+  _formState: FormState,
+  id: string
+) => ({
+  projectRef: {
+    id,
+  },
+});
```

Comment:
These can be better typed using the new TS `satisfies` operator, which applies types to the parameters and return value. There's a good writeup [here](https://www.learningtypescript.com/articles/the-satisfies-operator#real-world-usage-nextjs) about how it can apply to functions. I've used it for SpruceForm `validate` functions: https://github.com/evergreen-ci/spruce/blob/d1f969d680216e2ca45a73d74db380732a4d6169/src/pages/projectSettings/tabs/VariablesTab/VariablesTab.tsx#L91-L104

Reply:
thanks Sophie 🫶 please correct me if I used it wrong!